pandas/io/json/_normalize.py

```diff
@@ -307,6 +306,13 @@ def _recursive_extract(data, path, seen_meta, level=0):
                 for val, key in zip(meta, meta_keys):
                     if level + 1 > len(val):
```

Comment:
can you add some commentary on these branches (e.g. what cases they are handling)

Reply:
Will do